mbuild/recipes/silica_interface.py

```diff
@@ -0,0 +1,143 @@
+            below the surface to ensure the interface is coated.
+        """
+
+        tile_z = int(ceil((thickness + 2*O_buffer) / bulk_silica.periodicity[2]))
+        bulk = mb.TiledCompound(bulk_silica, n_tiles=(tile_x, tile_y, tile_z))
+
+        interface = mb.Compound(periodicity=(bulk.periodicity[0],
+                                             bulk.periodicity[1],
+                                             0.0))
+        for i, particle in enumerate(bulk.particles()):
+            if ((particle.name == 'Si' and O_buffer < particle.pos[2] < (thickness + O_buffer)) or 
+                    (particle.name == 'O' and particle.pos[2] < (thickness + 2*O_buffer))):
+                interface_particle = mb.Compound(name=particle.name, pos=particle.pos)
+                interface.add(interface_particle, particle.name + "_{}".format(i))
+        self.add(interface)
+
+    def _strip_stray_atoms(self):
+        """ Remove stray atoms and surface pieces """
+
+        major_component = max(nx.connected_components(self.bond_graph), key=len)
+        for atom in list(self.particles()):
+            if not self.bond_graph.has_node(atom):
+                self.remove(atom)
+            elif atom not in major_component:
+                self.remove(atom)
+
+    def _bridge_dangling_Os(self, oh_density, thickness):
+        """ Create Si-O-Si bridges on the surface to yield the desired
+            density of reactive surface sites
+        """
+
+        area = self.periodicity[0] * self.periodicity[1]
+        target = int(oh_density * area)
+
+        dangling_Os = []
+        for atom in list(self.particles()):
+            if atom.name == 'O' and atom.pos[2] > thickness and len(self.bond_graph.neighbors(atom)) == 1:
+                dangling_Os.append(atom)
+
+        n_bridges = int((len(dangling_Os) - target) / 2)
+
+        for _ in range(n_bridges):
+            bridged = False
+            while not bridged:
+                O1 = choice(dangling_Os)
+                Si1 = self.bond_graph.neighbors(O1)[0]
+                for O2 in dangling_Os:
+                    if O2 == O1:
+                        continue
+                    Si2 = self.bond_graph.neighbors(O2)[0]
+                    if Si1 == Si2:
+                        continue
+                    if any(neigh in self.bond_graph.neighbors(Si2) for neigh in self.bond_graph.neighbors(Si1)):
+                        continue
+                    r = self.min_periodic_distance(Si1.pos, Si2.pos)
+                    if r < 0.45:
+                        bridged = True
+                        self.add_bond((O1, Si2))
+                        dangling_Os.remove(O1)
+                        dangling_Os.remove(O2)
+                        self.remove(O2)
+                        break
+
+    def _identify_surface_sites(self, thickness):
+        """ Label surface sites and add ports above them """
+
+        for atom in list(self.particles()):
```

Comment:
I _think_ this `list()` casting is also unnecessary since we're ignoring ports in this loop but please double check.